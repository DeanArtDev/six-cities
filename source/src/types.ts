export enum AppState {
  PENDING = `pending`,
  READY = `ready`,
  LOADING = `loading`,
  SENT = `sent`,
}
export enum ErrorType {
  FAVORITE = `favorite`,
  UPLOAD_COMMENT = `uploadComment`,
  SING_IN = `singIn`,
}
export enum Authorization {
  AUTH = `AUTH`,
  NOT_AUTH = `NOT_AUTH`,
}
export enum PlacesClass {
  CITIES_LIST = `cities__places-list`,
  NEAR_LIST = `near-places__list`,
  CITIES_CARD = `cities__place-card`,
  CITIES_WRAPPER = `cities__image-wrapper`,
  NEAR_CARD = `near-places__card`,
  NEAR_WRAPPER = `near-places__image-wrapper`,
}
export enum OffersSortTypes {
  POPULAR = `Popular`,
  LOW_TO_HIGH = `Price: low to high`,
  HIGH_TO_LOW = `Price: high to low`,
  RATED_FIRST = `Top rated first`,
}
export enum NameSpace {
  MAIN = `MAIN`,
  USER = `USER`,
  DATA = `DATA`,
}
export enum FavoriteStatus {
  ON = `1`,
  OFF = `0`,
}
export enum ICON_SIZE {
  WIDTH = 30,
  HEIGHT = 40
}

export type Offer = {
  id: number,
  title: string,
  typeHousing: string,
  price: number,
  isPremium: boolean,
  features: Array<string>,
  maxAdults: number,
  bedrooms: number,
  isFavorite: boolean,
  host: {
    avatarUrl: string,
    id: number,
    isPro: boolean,
    name: string,
  },
  city: {
    coordinates: Array<number>,
    zoom: number,
    name: string,
  },
  location: {
    coordinates: Array<number>,
    zoom: number,
  },
  description: string,
  rating: number,
  images: Array<string>,
  prevImage: string,
  isProStatus: boolean,
}
export type Errors = {
  [key: string]: string | null,
}
export type AuthInfo = {
  id: number,
  email: string,
  name: string,
  avatarUrl: string,
  isProp: boolean,
}
export type SettingMap = {
  cityCoordinates: Array<number>,
  zoom: number,
}
export type CommentType = {
  id: number,
  rating: number,
  date: string,
  comment: string,
  user: {
    avatarUrl: string,
    name: string,
    id: number,
    isPro: boolean,
  },
}
export type FavoriteOffers = {
  [key: string]: Array<Offer>,
}
export type Id = number | null
