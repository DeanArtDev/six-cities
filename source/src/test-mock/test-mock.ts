const mockOffers = [
  {
    id: 333,
    title: `Beautiful & luxurious apartment at great location`,
    typeHousing: `apartment`,
    price: 200,
    isPremium: false,
    features: [`Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    bedrooms: 2,
    maxAdults: 4,
    isFavorite: true,
    host: {
      avatarUrl: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
      id: 25,
      isPro: true,
      name: `Angelina`,
    },
    city: {
      coordinates: [48.85661, 2.351499],
      zoom: 13,
      name: `Paris`,
    },
    location: {
      coordinates: [52.3909553943508, 4.929309666406198],
      zoom: 16,
    },
    description: `Lorem ipsum dolor sit amet, consectetur adipisicing elit.
      Aliquam aut distinctio ex recusandae. Debitis fugiat inventore itaque magni mollitia pariatur reprehenderit sapiente ullam.
      Cumque ducimus, est iusto nam natus neque nostrum quae veniam. A culpa eaque in laboriosam odio repellendus sequi sit voluptates.
      Ad adipisci architecto assumenda at autem consectetur consequuntur culpa, cumque debitis earum et eum ipsa iure nam,
      officiis placeat, quae qui quidem ratione rem tempore ut voluptate.`,
    rating: 1.7,
    images: [`apartment-01.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `room.jpg`, `studio-photos.jpg`, `studio-01.jpg`],
    prevImage: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
    isProStatus: false,
  },
  {
    id: 777,
    title: `Beautiful & luxurious apartment at great location`,
    typeHousing: `apartment`,
    price: 200,
    isPremium: false,
    features: [`Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    bedrooms: 2,
    maxAdults: 4,
    isFavorite: true,
    host: {
      avatarUrl: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
      id: 25,
      isPro: true,
      name: `Angelina`,
    },
    city: {
      coordinates: [48.85661, 2.351499],
      zoom: 13,
      name: `Paris`,
    },
    location: {
      coordinates: [52.3909553943508, 4.929309666406198],
      zoom: 16,
    },
    description: `Lorem ipsum dolor sit amet, consectetur adipisicing elit.
      Aliquam aut distinctio ex recusandae. Debitis fugiat inventore itaque magni mollitia pariatur reprehenderit sapiente ullam.
      Cumque ducimus, est iusto nam natus neque nostrum quae veniam. A culpa eaque in laboriosam odio repellendus sequi sit voluptates.
      Ad adipisci architecto assumenda at autem consectetur consequuntur culpa, cumque debitis earum et eum ipsa iure nam,
      officiis placeat, quae qui quidem ratione rem tempore ut voluptate.`,
    rating: 1.7,
    images: [`apartment-01.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `room.jpg`, `studio-photos.jpg`, `studio-01.jpg`],
    isProStatus: false,
    prevImage: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
  },
];

export const TestMock = {
  offers: [
    {
      city: {
        name: `Brussels`,
        location: {
          latitude: 50.846557,
          longitude: 4.351697,
          zoom: 13
        }
      },
      // eslint-disable-next-line camelcase
      preview_image: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg`,
      images: [`https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg`],
      title: `Waterfront with extraordinary view`,
      // eslint-disable-next-line camelcase
      is_favorite: false,
      // eslint-disable-next-line camelcase
      is_premium: false,
      rating: 3.7,
      type: `hotel`,
      bedrooms: 3,
      // eslint-disable-next-line camelcase
      max_adults: 6,
      price: 109,
      goods: [`Towels`, `Washer`, `Air conditioning`, `Cable TV`, `Washing machine`, `Fridge`, `Breakfast`],
      host: {
        id: 25,
        name: `Angelina`,
        // eslint-disable-next-line camelcase
        is_pro: true,
        // eslint-disable-next-line camelcase
        avatar_url: `img/avatar-angelina.jpg`
      },
      // eslint-disable-next-line camelcase
      id: 33,
      name: `Angelina`,
      // eslint-disable-next-line camelcase
      is_pro: true,
      // eslint-disable-next-line camelcase
      avatar_url: `img/avatar-angelina.jpg`,
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      location: {
        latitude: 50.837557,
        longitude: 4.339697,
        zoom: 16
      },
      latitude: 50.837557,
      longitude: 4.339697,
      zoom: 16,
    },
    {
      city: {
        name: `Paris`,
        location: {
          latitude: 50.846557,
          longitude: 4.351697,
          zoom: 13
        }
      },
      // eslint-disable-next-line camelcase
      preview_image: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg`,
      images: [`https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg`],
      title: `Waterfront with extraordinary view`,
      // eslint-disable-next-line camelcase
      is_favorite: false,
      // eslint-disable-next-line camelcase
      is_premium: false,
      rating: 3.7,
      type: `hotel`,
      bedrooms: 3,
      // eslint-disable-next-line camelcase
      max_adults: 6,
      price: 109,
      goods: [`Towels`, `Washer`, `Air conditioning`, `Cable TV`, `Washing machine`, `Fridge`, `Breakfast`],
      host: {
        id: 25,
        name: `Angelina`,
        // eslint-disable-next-line camelcase
        is_pro: true,
        // eslint-disable-next-line camelcase
        avatar_url: `img/avatar-angelina.jpg`
      },
      // eslint-disable-next-line camelcase
      id: 22,
      name: `Angelina`,
      // eslint-disable-next-line camelcase
      is_pro: true,
      // eslint-disable-next-line camelcase
      avatar_url: `img/avatar-angelina.jpg`,
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      location: {
        latitude: 50.837557,
        longitude: 4.339697,
        zoom: 16
      },
      latitude: 50.837557,
      longitude: 4.339697,
      zoom: 16,
    },
  ],
  adaptedOffers: [
    {
      id: 333,
      title: `Beautiful & luxurious apartment at great location`,
      typeHousing: `apartment`,
      price: 200,
      isPremium: false,
      features: [`Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
      bedrooms: 2,
      maxAdults: 4,
      isFavorite: true,
      host: {
        avatarUrl: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
        id: 25,
        isPro: true,
        name: `Angelina`,
      },
      city: {
        coordinates: [48.85661, 2.351499],
        zoom: 13,
        name: `Paris`,
      },
      location: {
        coordinates: [52.3909553943508, 4.929309666406198],
        zoom: 16,
      },
      description: `Lorem ipsum dolor sit amet, consectetur adipisicing elit.
      Aliquam aut distinctio ex recusandae. Debitis fugiat inventore itaque magni mollitia pariatur reprehenderit sapiente ullam.
      Cumque ducimus, est iusto nam natus neque nostrum quae veniam. A culpa eaque in laboriosam odio repellendus sequi sit voluptates.
      Ad adipisci architecto assumenda at autem consectetur consequuntur culpa, cumque debitis earum et eum ipsa iure nam,
      officiis placeat, quae qui quidem ratione rem tempore ut voluptate.`,
      rating: 1.7,
      images: [`apartment-01.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `room.jpg`, `studio-photos.jpg`, `studio-01.jpg`],
      prevImage: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
      isProStatus: false,
    },
    {
      id: 777,
      title: `Beautiful & luxurious apartment at great location`,
      typeHousing: `apartment`,
      price: 200,
      isPremium: false,
      features: [`Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
      bedrooms: 2,
      maxAdults: 4,
      isFavorite: true,
      host: {
        avatarUrl: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
        id: 25,
        isPro: true,
        name: `Angelina`,
      },
      city: {
        coordinates: [48.85661, 2.351499],
        zoom: 13,
        name: `Paris`,
      },
      location: {
        coordinates: [52.3909553943508, 4.929309666406198],
        zoom: 16,
      },
      description: `Lorem ipsum dolor sit amet, consectetur adipisicing elit.
      Aliquam aut distinctio ex recusandae. Debitis fugiat inventore itaque magni mollitia pariatur reprehenderit sapiente ullam.
      Cumque ducimus, est iusto nam natus neque nostrum quae veniam. A culpa eaque in laboriosam odio repellendus sequi sit voluptates.
      Ad adipisci architecto assumenda at autem consectetur consequuntur culpa, cumque debitis earum et eum ipsa iure nam,
      officiis placeat, quae qui quidem ratione rem tempore ut voluptate.`,
      rating: 1.7,
      images: [`apartment-01.jpg`, `apartment-02.jpg`, `apartment-03.jpg`, `room.jpg`, `studio-photos.jpg`, `studio-01.jpg`],
      isProStatus: false,
      prevImage: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
    },
  ],
  comments: [
    {
      id: 1,
      user: {
        id: 17,
        // eslint-disable-next-line camelcase
        is_pro: false,
        name: `Emely`,
        // eslint-disable-next-line camelcase
        avatar_url: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/8.jpg`,
      },
      rating: 5,
      comment: `The room was spacious and clean. The pool looked nothing e sauna and spa were closed for lunar new year holiday.`,
      date: `2020-02-12T03:29:40.085Z`,
    }
  ],
  adaptedComments: [
    {
      id: 1,
      rating: 3,
      date: `2020-02-13T01:42:13.129Z`,
      comment: `We loved it so much, the house, the veiw, the location`,
      user: {
        avatarUrl: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg`,
        name: `Mollie`,
        id: 16,
        isPro: false,
      },
    },
    {
      user: {
        id: 13,
        isPro: true,
        name: `Dean`,
        avatarUrl: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg`,
        avatarUrl2: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/7.jpg`,
      },
      id: 5,
      rating: 1,
      date: `2020-02-13T01:42:13.129Z`,
      comment: `We loved it so much, the house, the veiw, the location`,
    },
  ],
  cities: [`Paris`, `Brussels`, `Hamburg`, `Amsterdam`, `Cologne`],
  coordinates: [
    [52.3909553943508, 4.929309666406198],
    [52.3909553943508, 4.929309666406198],
    [52.3909553943508, 4.929309666406198],
    [52.3909553943508, 4.929309666406198],
  ],
  settingMap: {
    cityCoordinates: [52.3909553943508, 4.929309666406198],
    zoom: 16
  },
  authorizationStatus: {
    AUTH: `AUTH`,
    NOT_AUTH: `NOT_AUTH`,
  },
  authInfo: {
    id: 313,
    email: `email@email.com`,
    name: `You`,
    avatarUrl: `/avatar.svg`,
    isPro: true,
  },
  authInfoNotAdapted: {
    id: 313,
    email: `email@email.com`,
    name: `You`,
    // eslint-disable-next-line camelcase
    avatar_url: `/avatar.svg`,
    // eslint-disable-next-line camelcase
    is_pro: true,
  },
  favoriteOffers: {
    'Paris': mockOffers,
    'Moscow': mockOffers,
  },
  commentPost: {
    "comment": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    "rating": 4
  }
};
