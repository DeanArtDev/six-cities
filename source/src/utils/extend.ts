const extend = (a: object, b: object): object => {
  const emptyObj: object = {};

  return Object.assign(emptyObj, a, b);
};

export default extend;
