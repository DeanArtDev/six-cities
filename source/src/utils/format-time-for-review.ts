import {MONTHS} from '@root/consts/consts';

const formatTimeForReview = (date: string): string => {
  const nextDate = new Date(date);
  const month = MONTHS[nextDate.getMonth()];
  const day = nextDate.getDate() < 10 ? `0${nextDate.getDate()}` : nextDate.getDate();

  return `${month} ${day} ${nextDate.getFullYear()}`;
};

export default formatTimeForReview;
