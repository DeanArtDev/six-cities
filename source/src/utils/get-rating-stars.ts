const ONE_STAR_PRESENTS = 20;

const getRatingStars = (value: number): number | 0 => {
  if (value === 0 || value === undefined || isNaN(value)) {
    return 0;
  }

  return Math.ceil(value) * ONE_STAR_PRESENTS;
};

export default getRatingStars;
