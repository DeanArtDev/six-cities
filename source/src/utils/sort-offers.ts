import {OffersSortTypes} from '@root/types';

const sortOffers = (typeSort, offers) => {
  switch (typeSort) {
    case OffersSortTypes.LOW_TO_HIGH:
      return offers
        .slice()
        .sort((max, min) => max.price - min.price);

    case OffersSortTypes.HIGH_TO_LOW:
      return offers
        .slice()
        .sort((max, min) => min.price - max.price);

    case OffersSortTypes.RATED_FIRST:
      return offers
        .slice()
        .sort((max, min) => min.rating - max.rating);

    default:
      return offers;
  }
};

export default sortOffers;
