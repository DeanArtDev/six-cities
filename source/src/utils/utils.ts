export {default as getRatingStars} from './get-rating-stars';
export {default as extend} from './extend';
export {default as sortOffers} from './sort-offers';
export {default as formatTimeForReview} from './format-time-for-review';
export {default as noop} from './noop';
