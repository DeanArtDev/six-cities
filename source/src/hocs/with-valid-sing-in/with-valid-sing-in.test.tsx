import withValidSingIn from './with-valid-sing-in';
import {AppState} from '@root/types';
import {Authorization} from '@root/types';
import {TestMock} from '@root/test-mock/test-mock';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {noop} from '@root/utils/utils';

describe(`<withValidSingIn/>: `, () => {

  const WrappedComponent = React.forwardRef(
      function mockComponent(props, ratingRef: React.LegacyRef<HTMLElement>) {
        return (
          <section ref={ratingRef}/>
        );
      }
  );

  const MockWrapper = withValidSingIn(WrappedComponent);

  describe(`App State: `, () => {
    it(`should render READY correctly`, () => {
      const wrapper = renderer
        .create(
            <MockWrapper
              currentCity={TestMock.cities[0]}
              appState={AppState.READY}
              authStatus={Authorization.AUTH}
              onSubmit={noop}
              checkAuth={noop}
              errors={null}
            />,
            {createNodeMock: () => document.createElement(`section`)}
        ).toJSON();

      expect(wrapper).toMatchSnapshot();
    });
    it(`should render LOADING correctly`, () => {
      const wrapper = renderer
        .create(
            <MockWrapper
              currentCity={TestMock.cities[0]}
              appState={AppState.LOADING}
              authStatus={Authorization.AUTH}
              onSubmit={noop}
              checkAuth={noop}
              errors={null}
            />,
            {createNodeMock: () => document.createElement(`section`)}
        ).toJSON();

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe(`Errors: `, () => {
    it(`should render error correctly`, () => {
      const wrapper = renderer
        .create(
            <MockWrapper
              currentCity={TestMock.cities[0]}
              appState={AppState.LOADING}
              authStatus={Authorization.AUTH}
              onSubmit={noop}
              checkAuth={noop}
              errors={`errors`}
            />,
            {createNodeMock: () => document.createElement(`section`)}
        ).toJSON();

      expect(wrapper).toMatchSnapshot();
    });

  });

});
