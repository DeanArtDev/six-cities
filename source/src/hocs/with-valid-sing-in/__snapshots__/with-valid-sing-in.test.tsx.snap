// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<withValidSingIn/>:  App State:  should render LOADING correctly 1`] = `<section />`;

exports[`<withValidSingIn/>:  App State:  should render READY correctly 1`] = `<section />`;

exports[`<withValidSingIn/>:  Errors:  should render error correctly 1`] = `<section />`;
